atom/browser/lib/init.coffee

```diff
@@ -9,6 +9,27 @@ process.argv.splice 1, 1
 
 # Add browser/api/lib to require's search paths,
 # which contains javascript part of Atom's built-in libraries.
+if process.platform is 'win32'
+  homeDir = process.env.USERPROFILE
+else
+  homeDir = process.env.HOME
+
+syspath = []
+
+if homeDir
+  syspath.push(path.resolve(homeDir, '.node_modules'))
+  syspath.push(path.resolve(homeDir, '.node_libraries'))
+
+nodePath = process.env['NODE_PATH']
```

Comment:
We should not remove paths in `$NODE_PATH`, some applications are using it to customize module search paths.